hubclient/client.go

```diff
@@ -39,6 +39,7 @@ type Client struct {
 	haveCsrfToken bool
 	csrfToken     string
 	debugFlags    HubClientDebug
+	headerValues  map[string]string
```

Comment:
https://golang.org/pkg/net/http/#Header is more appropriate type to use

Reply:
Sorry I did not make myself clear. It should still be additive, but to change this storage for defaults changes from `map[string]string` to `http.Header`. It would be cleaner to have `headerValues` (maybe it should be `additionalHeaderValues`, or `overrideHeaderValues`, or just `additionalHeader` -- whatever the intent is) stored in something that follows the same rules as http.Header that will be used later for each request. "Adding a header value" is not the same as "Setting header value" - first is additive (header key may hold multiple values, like an array), second has a 'replace' semantic in it. 
You don't have to expose the whole `header`, just Add, Set, and maybe Delete as AddHeaderValue, SetHeaderValue, and DeleteHeaderValue. If you follow "non-exposing" path, you actually can move csrfToken right into it, but that would require more changes.

Reply:
Updated to use a header type here and exposed add in addition to set/delete, updated the logic to ensure it's an override and not an additive mechanism